src/dataconnect/schemaMigration.ts

```diff
@@ -441,14 +486,30 @@ async function ensureServiceIsConnectedToCloudSql(
   }
 }
 
-function displaySchemaChanges(error: IncompatibleSqlSchemaError) {
+function displaySchemaChanges(
+  error: IncompatibleSqlSchemaError,
+  schemaValidation: SchemaValidation | "STRICT_AFTER_COMPATBILE",
+) {
   switch (error.violationType) {
     case "INCOMPATIBLE_SCHEMA":
       {
-        const message =
-          "Your new schema is incompatible with the schema of your CloudSQL database. " +
-          "The following SQL statements will migrate your database schema to match your new Data Connect schema.\n" +
-          error.diffs.map(toString).join("\n");
+        let message;
+        if (schemaValidation === "COMPATIBLE") {
+          message =
+            "Your new schema is incompatible with the schema of your CloudSQL database. " +
```

Comment:
Consider including the actual instance/database ID in this error message - as we move toward multiple data sources, we should make sure to be explicit about where changes will be made